src/services/transaction/TransactionSubmitService.ts

```diff
@@ -33,14 +33,7 @@ export class TransactionSubmitService extends AbstractService {
 		try {
 			tx = api.tx(transaction);
 		} catch (err) {
-			const { cause, stack } = extractCauseAndStack(err);
-
-			throw {
-				error: 'Failed to parse transaction.',
-				transaction,
-				cause,
-				stack,
-			};
+			throw new BadRequest('Failed to parse transaction.');
```

Comment:
is there a way we can continue including/returning the transaction/stack information as part of the error?  This info can useful when looking into specific issues/debugging.

Reply:
Plus one to this, otherwise it looks really great!

Reply:
Yes exactly! 💯 Thank you both for pointing this out 🙏 I forgot to mention it in the PR description. 
I tried to find a way to keep the stack and cause from `extractCauseAndStack` for this exact reason that you mention but I was still getting a `500` error at the end. I will check some more to see how I can override the `StatusCode` returned. Thank YOU both for your time on reviewing !  🙏

Reply:
Thank you @TarikGul  and @marshacb 🙏  to : 
- point out the fact that we prefer keeping the debugging info and
- point in the direction of the `middleware` files for the error handling 

Based on those sugegstions, I removed the `BadRequest` solution and opted for the solution of passing the status code in the `txErrorMiddleware`. 
This implied the following changes : 
- Reverted to the previous error handling with `extractCauseAndStack`
- Added the `code` in `err` and passing it in the `txErrorMiddleware`
- Reverted the changes to the test `.spec` files
- Updated the mock files with the error code returned since now `err` also includes the `code`
- Updated `ITxLegacyError`
- Reverted changes in docs and rebuilt

**Important Note**
- I am not setting a default value for `code` in the `txErrorMiddleware` so if it is not set in the service file or from wherever it is called then it will result in a `500` status code and it will output a not so pretty ` error: Invalid status code: undefined`. Maybe we would like to handle this more gracefully ?